src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/UpdateIndexWorkitemInstanceCore.sql

```diff
@@ -0,0 +1,114 @@
+/***************************************************************************************/
+-- STORED PROCEDURE
+--    UpdateIndexWorkitemInstanceCore
+--
+-- DESCRIPTION
+--    Updates workitem query tag values.
+--    
+-- PARAMETERS
+--     @workitemKey
+--         * Refers to WorkItemKey
+--     @stringExtendedQueryTags
+--         * String extended query tag data
+--     @dateTimeExtendedQueryTags
+--         * DateTime extended query tag data
+--     @personNameExtendedQueryTags
+--         * PersonName extended query tag data
+-- RETURN VALUE
+--     None
+/***************************************************************************************/
+CREATE OR ALTER PROCEDURE dbo.UpdateIndexWorkitemInstanceCore
+    @workitemKey                                                                 BIGINT,
+    @stringExtendedQueryTags dbo.InsertStringExtendedQueryTagTableType_1         READONLY,
+    @dateTimeExtendedQueryTags dbo.InsertDateTimeExtendedQueryTagTableType_2     READONLY,
+    @personNameExtendedQueryTags dbo.InsertPersonNameExtendedQueryTagTableType_1 READONLY
+AS
+BEGIN
+
+    DECLARE @workitemResourceType TINYINT = 1
+    DECLARE @newWatermark BIGINT
+
+    SET @newWatermark = NEXT VALUE FOR dbo.WatermarkSequence
+
+    -- String Key tags
+    IF EXISTS (SELECT 1 FROM @stringExtendedQueryTags)
+    BEGIN
+
+        UPDATE ets
```

Comment:
Do we need 2 UPDATES? Could you check if this solves

        UPDATE ets
		SET    TagValue = CASE
			WHEN ets.TagValue <> input.TagValue
				THEN input.TagValue
			END,
            Watermark = @newWatermark
        FROM dbo.ExtendedQueryTagString AS ets
        INNER JOIN @stringExtendedQueryTags AS input
            ON ets.TagKey = input.TagKey
        WHERE
            SopInstanceKey1 = @workitemKey
            AND ResourceType = @workitemResourceType

Reply:
This causes COLLATE issue when comparing TagValue. And even with explicit COLLATE being mentioned, the update fails with Unique Index constraint error, as it gets triggered with the updates (with the same value). Explicit 2 step update is the only solution that is working for now. Other option to consider is if we just want to update watermark only when the TagValue is updated, we could have it as a single UPDATE.

Reply:
As per discussion changed the query to update watermark only for the tag records that are updated with the new value.